### DISCUSSION TITLE
VanJS + BlueprintJS + Vite + Typescript

### DISCUSSION THREAD

**Discussion by iongion** [vanjs-org/van#24] [Show and tell]
I am hacking around this to learn VanJS - nothing is "production" grade.

Blueprint is a high quality react-typescript UI framework - <https://blueprintjs.com/docs/>

https://stackblitz.com/edit/vitejs-vite-kc1gr9

It is a startup sample, what I am following

1. A way to build idiomatic VanJS "components"
2. Understand reactivity - the vanjs way
3. Port as many components of BlueprintJS framework as possible, deal with changes of properties over time
4. Introduce a svg compiler for vite that transforms svgs into vanjs syntax - prior art <https://www.npmjs.com/package/vite-svg-loader> for vue or <https://www.npmjs.com/package/vite-plugin-solid-svg> for solidjs or even <https://www.npmjs.com/package/vite-plugin-svgr> for react
5. Introduce animations
6. Complex state / store management
7. Reactive translations a-la gettext / i18n
8. Reactive routing a-la SPA
9. Reactive Auth0 authentication
10. Keep it as close to JS as possible 
11. Port CSS per component instead of the global import to benefit bundling optimizations

**Comment by Tao-VanJS** [comment]
This is awesome! ✨

Thank you so much your interest in VanJS and sharing your learning experience with the community!

**Comment by Nakamcode** [comment]
Keep it coming @iongion, this is very insightful.
